### ISSUE TITLE
Module 'amplify_auth_cognito' not found in latest version of flutter and amplify #881 

### ISSUE THREAD

**Issue by richanshah** [aws-amplify/amplify-flutter#5705]
### Description

Description: Build succeeds in Xcode run and also runs fine in Simulator, But When Archiving the runner
Error : "Parse Issue Module 'amplify_auth_cognito' not found"

      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

[✓] Connected device (5 available)
    • HD1901 (mobile)                 • 6c346dec                             • android-arm64  • Android 12 (API 31)
    • iPhone 16 Plus (mobile)         • B120580B-673B-40C4-B1FD-254BA8CAA74D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-18-0 (simulator)
    • macOS (desktop)                 • macos                                • darwin-arm64   • macOS 15.1 24B83 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         • macOS 15.1 24B83 darwin-arm64
    • Chrome (web)                    • chrome                               • web-javascript • Google Chrome 131.0.6778.86
    ! Error: Browsing on the local area network for iPhone 14 White. Ensure the device is unlocked and attached with a cable or associated with the same local area
      network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
    ! Error: Browsing on the local area network for iPhone 13 Pro Blue. Ensure the device is unlocked and attached with a cable or associated with the same local
      area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
    ! Error: Browsing on the local area network for BV iPhone 11 black. Ensure the device is unlocked and attached with a cable or associated with the same local
      area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.
<img width="1260" alt="image" src="https://github.com/user-attachments/assets/646160c7-9427-4eba-bc7c-9352a0e7bc9a">

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [X] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

build with latest plugin and flutter and try flutter clean, pub get,cd ios,pod install/update and flutter build ios command

### Screenshots

<img width="1260" alt="image" src="https://github.com/user-attachments/assets/de0e9162-5947-4681-9179-a71d80eb619d">

### Platforms

- [X] iOS
- [ ] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.24.0

### Amplify Flutter Version

2.5.0

### Deployment Method

Amplify Gen 2

**Comment by ekjotmultani** [comment]
Hi @richanshah, I'm sorry you are facing this issue. We will investigate and attempt to reproduce this issue, and will get back to you when we have an update.

**Comment by richanshah** [comment]
Hi @ekjotmultani ,

I’ve identified the root cause of the issue, which was related to the build settings. Specifically, it was the Optimization Level configuration. Setting the optimization level to "None" (both for general build settings and the Swift compiler) resolved the issue.

Here are the steps I followed:

Set the Optimization Level to "None" in the build settings.
Set "No Optimization" in the Swift compiler settings.
Performed a flutter clean.
Removed the Pods and Podfile.lock.
Restarted the process and reinstalled the Pods.
Additionally, I reviewed the dependency versions and ensured they were updated as follows:

amplify_flutter: ^2.2.0 to 2.5.0
amplify_auth_cognito: ^2.2.0 to 2.5.0
amplify_core: ^2.2.0 to 2.5.0
Finally, I removed and re-added the required libraries in Link Binary with Libraries, which helped resolve the linking issues.
